src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/utils/v2/UserReceiptMailBuilder.kt

```diff
@@ -82,67 +83,108 @@ class UserReceiptMailBuilder(@Autowired private val confidentialDataUtils: Confi
     baseTransactionWithRequestedUserReceipt: BaseTransactionWithRequestedUserReceipt,
     emailAddress: String
   ): NotificationsServiceClient.SuccessTemplateRequest {
-    val language = "it-IT"
+    val transactionUserReceiptData =
+      baseTransactionWithRequestedUserReceipt.transactionUserReceiptData
     val transactionAuthorizationRequestData =
       baseTransactionWithRequestedUserReceipt.transactionAuthorizationRequestData
     val transactionAuthorizationCompletedData =
       baseTransactionWithRequestedUserReceipt.transactionAuthorizationCompletedData
-    val transactionUserReceiptData =
-      baseTransactionWithRequestedUserReceipt.transactionUserReceiptData
     return NotificationsServiceClient.SuccessTemplateRequest(
       to = emailAddress,
       language = transactionUserReceiptData.language,
       subject = "Il riepilogo del tuo pagamento",
       templateParameters =
-        SuccessTemplate(
-          TransactionTemplate(
-            baseTransactionWithRequestedUserReceipt.transactionId
-              .value()
-              .toString()
-              .lowercase(Locale.getDefault()),
-            dateTimeToHumanReadableString(
-              ZonedDateTime.parse(transactionUserReceiptData.paymentDate),
-              Locale.forLanguageTag(language)),
-            amountToHumanReadableString(
-              baseTransactionWithRequestedUserReceipt.paymentNotices
-                .stream()
-                .mapToInt { paymentNotice: PaymentNotice ->
-                  paymentNotice.transactionAmount().value()
-                }
-                .sum() + transactionAuthorizationRequestData.fee),
-            PspTemplate(
-              transactionAuthorizationRequestData.pspBusinessName,
-              FeeTemplate(amountToHumanReadableString(transactionAuthorizationRequestData.fee))),
-            transactionAuthorizationCompletedData.rrn
-              ?: transactionAuthorizationRequestData.authorizationRequestId,
-            // TODO: auth code is mandatory for OK mail template but not received for PAYPAL
-            // TODO task CHK-1970 will take cares of handle mail template for APM methods
-            transactionAuthorizationCompletedData.authorizationCode ?: "-",
-            PaymentMethodTemplate(
-              transactionAuthorizationRequestData.paymentMethodDescription,
-              transactionAuthorizationRequestData.transactionGatewayAuthorizationRequestedData.logo
-                .toString(),
+        when (transactionAuthorizationRequestData.paymentTypeCode) {
+          "PPAL",
```

Comment:
What about define the PaymentCode as a type or a set of constants? Something like
```kotlin
object PaymentCode {
    const val PAYPAL = "PPAL",
    const val MY_BANK = "MyBank"
    ...
}
```
or an enum

Reply:
I've created the Enum and used it in all the points of the code where it was needed:
https://github.com/pagopa/pagopa-ecommerce-event-dispatcher-service/blob/1a850cb5488801e12bd037bf6c3b7e7a1f46752c/src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/utils/PaymentCode.kt#L1-L13